app/experimenter/experiments/api/v5/types.py

```diff
@@ -309,13 +309,34 @@ def resolve_all_feature_configs(root, info):
         return NimbusFeatureConfig.objects.all().order_by("name")
 
     def resolve_firefox_versions(root, info):
-        return root._text_choices_to_label_value_list(NimbusExperiment.Version)
+        return NimbusConfigurationType.sort_version_choices(NimbusExperiment.Version)
 
     def resolve_conclusion_recommendations(root, info):
         return root._text_choices_to_label_value_list(
             NimbusExperiment.ConclusionRecommendation
         )
 
+    @staticmethod
+    def sort_version_choices(choices):
+        def version_key(x):
+            version_str = x
+            if not version_str:
+                return (1,)
+            return tuple(map(int, version_str.replace("!", "0").split(".")))
+
+        sorted_versions = sorted(choices, key=version_key, reverse=True)
+        sorted_versions = [
+            NimbusLabelValueType(
+                label=name.label,
+                value=name.label.replace(" ", "_").upper(),
+            )
+            for name in sorted_versions
+        ]
+        no_version_option = sorted_versions.pop()
+        sorted_versions.insert(0, no_version_option)
+
+        return sorted_versions
```

Comment:
1) Why isn't this using the [existing parsing method](https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/constants.py#L243-L245) that lets you sort by semantic versioning?

2) Why do we need to sort this list at all?  The list is static and already ordered, if all you want to do is reverse it you can just `[::-1]`?

Reply:
yeah for point 1, I could have used the parse method, totally missed that. For point 2, options are already ordered, reversing it is a quick option, I was overthinking if the list is not sorted in the future. I will do the second point reversing list and add no version as the first option, thanks @jaredlockhart